js/electron/transcoding.js

```diff
@@ -53,8 +127,48 @@ function transcodingFactory(electronIpcRenderer) {
       electronIpcRenderer.send('transcode-video-request', filePath);
     });
   }
+
+  return { downloadBinaries, transcode };
 }
 
+async function binariesDownloader(electronIpcMain, userDataFolder) {
+  const ffbinFolder = path.join(userDataFolder, FfbinDir);
+
+  const ffmpegPath = path.join(ffbinFolder, 'ffmpeg');
+  const ffprobePath = path.join(ffbinFolder, 'ffprobe');
+
+  const oneGbBytes = 1024 * 1024 * 1024;
+
+  const spaceCalc = await checkDiskSpace(ffbinFolder);
+  const isEnoughSpace = (spaceCalc.free > oneGbBytes);
+
+  if (!isEnoughSpace) {
+    const err = Error('NO_DISK_SPACE');
+    e.sender.send('transcode-binaries-error', err);
+    return;
+  }
+
+  electronIpcMain.on('transcode-binaries-request', async (e) => {
+    downloadFfBinaries(ffbinFolder, (progress) => {
+      console.log(`FF Binaries download`, progress);
+
+      e.sender.send('transcode-binaries-progress', progress);
+    }).then(() => {
+      console.log(`FF Binaries are downloaded`);
+
+      ffmpeg.setFfmpegPath(ffmpegPath);
+      ffmpeg.setFfprobePath(ffprobePath);
+
+      e.sender.send('transcode-binaries-progress', 100);
+      e.sender.send('transcode-binaries-ready');
+    }).catch((err) => {
+      console.error(err);
+      console.log('FF Binaries not downloaded. Proceeding as is...');
+
+      e.sender.send('transcode-binaries-error', err);
+    });
+  });
+}
```

Comment:
### FF Binaries download logic
Checks if there is free space for FFMPEG and FFPROBE. Assuming that we need at least 1 Gb for stable work of app.

This function is listening to events called from UI scripts too.